biblio-iacr.el

```diff
@@ -0,0 +1,91 @@
+;;; biblio-iacr.el --- description -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2020 Lorenzo Martinico
+
+;; Author: Lorenzo Martinico <lzmartinico@gmail.com>
+;; URL: https://github.com/cpitclaudel/biblio.el
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;; This file implements a backend for the for the `biblio' package (which see for more
+;; documentation).
+;;
+;;; Code:
+
+(require 'biblio-core)
+
+(defun biblio-iacr--forward-bibtex (metadata forward-to)
+  "Forward BibTeX for IACR entry METADATA to FORWARD-TO."
+    (funcall forward-to
+         (format "@misc{cryptoeprint:%s,\n    author = {%s},\n    title = {%s},\n    howpublished = {%s},\n    year = {%s},\n    note = {\\url{%s}},\n}"
+                 (biblio-alist-get 'eprintid metadata)
+                 (car (biblio-alist-get 'authors metadata))
+                 (biblio-alist-get 'title metadata)
+                 (biblio-alist-get 'container metadata)
+                 (biblio-alist-get 'year metadata)
+                 (biblio-alist-get 'url metadata))))
+
+(defun biblio-iacr--extract-interesting-fields (header body)
+  "Prepare a IACR search result, composed of HEADER and BODY, for display.
+HEADER contains the technical report id, which includes the year of publication.
+BODY includes the Title and Authors."
+  (let ((eprintid (caddr (nth 3 header))))
+        (list (cons 'year (substring eprintid 0 4))
+                (cons 'title (nth 2 (caddr body)))
+                (cons 'authors (list (nth 2 (nth 2 (nth 4 body)))))
+                (cons 'container (format "Cryptology ePrint Archive, Report %s" eprintid))
+                (cons 'type "eprint")
+                (cons 'references nil)
+                (cons 'eprintid eprintid)
+                (cons 'url (format "https://eprint.iacr.org/%s.pdf" eprintid)))))
+
+(defun biblio-iacr--traverse-results (list)
+  "Traverse the IACR html nodes LIST to produce a list of results recursively."
+         (if list
+                (cons (biblio-iacr--extract-interesting-fields (car list) (cadr list)) (biblio-iacr--traverse-results (cddr list)))
+                nil))
```

Comment:
Emacs lisp isn't very good at recursion; people traditionally do something like this instead:

```elisp
(let ((entries nil))
  (while input
    (push entries (extract-fields (car entries) (cadr entries)))
    (setq entries (cddr entries)))
  (nreverse entries))
```